lib/bricks-net-api-types.js

```diff
@@ -0,0 +1,1176 @@
+								_this.OnChunk(_this.buffer_.toString('utf8', chunk_offset, chunk_offset + chunk_length));
+								next_line_offset = next_offset;
+								
+								// after if (!first_line_parsed)
+								current_line_offset = next_line_offset;
+								
+								// continue while2
+								part2();
+							}, finish);
+							return;
+						}
+						_this._DEBUG_LOG('TemplatedHTTPRequestData: part5 offset >= next_offset OnChunk');
+						_this.OnChunk(_this.buffer_.toString('utf8', chunk_offset, chunk_offset + chunk_length));
+						next_line_offset = next_offset;
+					}
+				}
+			}
+			else if (!line_is_blank) {
+				_this._DEBUG_LOG('TemplatedHTTPRequestData: part5 else !line_is_blank',
+					'current_line_offset ===', current_line_offset);
+				
+				var p = strstr(_this.buffer_, kHeaderKeyValueSeparator, current_line_offset);
+				
+				_this._DEBUG_LOG('TemplatedHTTPRequestData: part5 else !line_is_blank',
+					'p ===', p);
+				
+				if (p >= 0) {
+					// We cannot end the string with '\0' so we have to track indexes.
+					//_this.buffer_[p] = '\0';
+					var key = _this.buffer_.slice(current_line_offset, p).toString('utf8');
+					var value = _this.buffer_.slice(p + kHeaderKeyValueSeparatorLength, next_crlf_ptr).toString('utf8');
+					
+					_this._DEBUG_LOG('TemplatedHTTPRequestData: part5 else !line_is_blank p >= 0 OnHeader',
+						'key ===', JSON.stringify(key),
+						'value ===', JSON.stringify(value));
+					
+					_this.OnHeader(key, value);
+					
+					if (key === kContentLengthHeaderKey) {
+						body_length = parseInt(value, 10);
+					}
+					else if (key === kTransferEncodingHeaderKey) {
+						if (value === kTransferEncodingChunkedValue) {
+							chunked_transfer_encoding = true;
+						}
+					}
+				}
+			}
+			else {
+				_this._DEBUG_LOG('TemplatedHTTPRequestData: part5 else');
+				
+				if (!chunked_transfer_encoding) {
+					_this._DEBUG_LOG('TemplatedHTTPRequestData: part5 else !chunked_transfer_encoding');
+					
+					body_offset = next_line_offset;
+					if (body_length !== UINT_MAX) {
+						length_cap = body_offset + body_length;
+						if (length_cap + 1 > _this.buffer_.length) {
+							var delta_size = length_cap + 1 - _this.buffer_.length;
+							var buffer_new_length = Math.min(delta_size, buffer_max_growth_due_to_content_length);
+							_this.buffer_ = Buffer.concat([
+								_this.buffer_, 
+								new Buffer(buffer_new_length - _this.buffer_.length)
+							], buffer_new_length);
+						}
+					}
+					else {
+						length_cap = body_offset;
+					}
+				}
+				else {
+					_this._DEBUG_LOG('TemplatedHTTPRequestData: part5 else -> receiving_body_in_chunks');
+					
+					receiving_body_in_chunks = true;
+				}
+			}
+			current_line_offset = next_line_offset;
+			
+			_this._DEBUG_LOG('TemplatedHTTPRequestData: part5 continue while2',
+				'current_line_offset ===', current_line_offset);
+			
+			// continue while3
+			part4();
```

Comment:
Yes, the overall canvas is the same. I had to rewrite the read, the buffers and the pointers logic, though.